pkg/cmd/admin/policy/psp_review.go

```diff
@@ -0,0 +1,302 @@
+	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", "z", o.ServiceAccount, "service account in the current namespace to use as a user")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+// Related to PodSecurityPolicyReview
+var (
+	reviewLong = templates.LongDesc(`Checks which ServiceAccount can create a Pod.
+	The Pod is inferred from the PodTemplateSpec in the provided resource.
+	If no ServiceAccount is provided the ServiceAccount specified in podTemplateSpec.spec.serviceAccountName is used,
+	unless the podTemplateSpec.spec.serviceAccountName is empty, in which case "default" is used.
+	If ServiceAccounts are provided the podTemplateSpec.spec.serviceAccountName is ignored.`)
+	reviewExamples = templates.Examples(`# Check whether ServiceAccount sa1 and sa2 can admit a Pod with TemplatePodSpec specified in my_resource.yaml
+	# ServiceAccount specified in myresource.yaml file is ignored
+	$ %[1]s -s sa1,sa2 -f my_resource.yaml
+	
+	# Check whether ServiceAccount specified in my_resource_with_sa.yaml can admit the Pod
+	$ %[1]s -f my_resource_with_sa.yaml
+	
+	# Check whether default ServiceAccount can admit the Pod, default is taken since no ServiceAccount is defined in myresource_with_no_sa.yaml
+	$  %[1]s -f myresource_with_no_sa.yaml
+	`)
+)
+
+const ReviewRecommendedName = "review"
+
+type reviewOptions struct {
+	client    client.PodSecurityPolicyReviewsNamespacer
+	Namespace string
+
+	PodTemplateSpec     *kapi.PodTemplateSpec
+	ServiceAccountNames []string
+	Filename            string
+	AllServiceAccounts  bool
+
+	PrintObject func(obj runtime.Object, out io.Writer) error
+}
+
+func (o *reviewOptions) Run(out io.Writer) error {
+	review := &securityapi.PodSecurityPolicyReview{
+		Spec: securityapi.PodSecurityPolicyReviewSpec{
+			Template:            *o.PodTemplateSpec,
+			ServiceAccountNames: o.ServiceAccountNames,
+		},
+	}
+
+	response, err := o.client.PodSecurityPolicyReviews(o.Namespace).Create(review)
+	if err != nil {
+		return fmt.Errorf("unable to compute PSP rviews: %v", err)
+	}
+	return o.PrintObject(response, out)
+}
+
+func (o *reviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command) error {
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	oclient, _, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.client = oclient
+	o.Namespace = namespace
+	mapper, typer := f.Object()
+
+	infos, err := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(namespace).
+		FilenameParam(enforceNamespace, &resource.FilenameOptions{Recursive: false, Filenames: []string{o.Filename}}).
+		SingleResourceType().
+		Do().
+		Infos()
+
+	if err != nil {
+		return fmt.Errorf("unable to find a resource: %v", err)
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("expected a resource for path: %s", o.Filename)
```

Comment:
OK, I understood exactly the opposite. Only one resource and no `-R` :)
Anyway, reworking it to have multiple resources.